linphone: 4.3.2 -> 4.4.1

ThibaultLemaire opened the pull request:
###### Description of changes

https://linphone.org/news/new-releases-linphone-apps-and-linphone-sdk

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

ThibaultLemaire committed:
linphone: 4.2.5 -> 4.4.1

Co-authored-by: Lorenz Brun <lorenz@brun.one>
Co-authored-by: Austin Butler <abutler@riotgames.com>

lorenz commented on pkgs/development/libraries/decaf/default.nix:
```diff
@@ -0,0 +1,38 @@
+  # Do not build static libraries
+  cmakeFlags = [ "-DENABLE_STATIC=NO" ];
+
+  NIX_CFLAGS_COMPILE = [ ];
```
Can't you leave this line out?

ThibaultLemaire replied:
Well, it turns out I had to keep it because I hit some warnings (interpreted as errors) when rebasing

lorenz commented on pkgs/development/libraries/decaf/default.nix:
```diff
@@ -0,0 +1,38 @@
+  ];
+
+  src = fetchFromGitLab {
+    domain = "gitlab.linphone.org";
```
If you're pulling decaf from your GitLab this should IMO be called bc-decaf or something. You're not packaging upstream decaf.

lorenz commented on pkgs/development/libraries/lime/default.nix:
```diff
@@ -3,33 +3,44 @@
-  version = "5.0.66";
+  version = "5.1.12";
 
+  #  src = /home/tlemaire/projects/linphone-desktop/linphone-sdk/lime;
```
Left-over

lorenz commented:
I left some comments on some specific things. Overall this looks fine. Do you know what changes are in the Belledonne fork of soci making it necessary to use that one over upstream?

ThibaultLemaire commented:
Oops, that rebase pulled in a lot of commits...

> Do you know what changes are in the Belledonne fork of soci making it necessary to use that one over upstream?

I can't tell you much more than what a git diff would. I know we are one major version behind, with some cherry-picked commits. We have ongoing efforts to rebase on top of 4.0, but no plans to submit our changes upstream (to my knowledge).

As to why it's necessary: The linphone desktop I compiled was unable to display any chat history or contact list, and segfaulted approximately one second after receiving any chat message (I didn't time it, but it was reproducible).

ThibaultLemaire committed:
Cleanup linphone dependencies

Through trial and error: delete all dependencies, try to build, try to
run, reintroduce missing dependencies.

It's possible that removing some of those silently disabled some features
that I didn't see (codecs, encryption schemes, etc...)

lorenz commented:
@ThibaultLemaire I'm closing this as the primary part of it has been merged upstream. Your dependency cleanup was not included yet because it got merged before I had time to add it. I'd recommend you submit a separate PR to the main nixpkgs repo with it.

ThibaultLemaire commented:
@lorenz https://github.com/NixOS/nixpkgs/pull/171368
